src/factory.js

```diff
@@ -120,33 +123,54 @@ class FSPIOPError extends MojaloopFSPIOPError {
         e.errorInformation.extensionList.extension = _.cloneDeep(this.extensions.extension)
       }
 
-      // TODO: Need to clarify ML API Specification for the correct model structure for the extensionList - catering for both scenarios until this can be clarified
-      // const causeKeyValueFromExtensions = e.errorInformation.extensionList.find(keyValue => keyValue.key === 'cause')
-      const causeKeyValueFromExtensions = e.errorInformation.extensionList.extension.find(keyValue => keyValue.key === 'cause')
-      if (causeKeyValueFromExtensions) {
-        causeKeyValueFromExtensions.value = `${this.stack}\n${causeKeyValueFromExtensions.value}` // truncate string to match Mojaloop API v1.0 Spec
-      } else {
-        const causeKeyValue = {
-          key: 'cause',
-          value: this.stack // truncate string to match Mojaloop API v1.0 Spec
+      if (includeCauseExtension) {
+        // TODO: Need to clarify ML API Specification for the correct model structure for the extensionList - catering for both scenarios until this can be clarified
+        const causeKeyValueFromExtensions = e.errorInformation.extensionList.extension.find(keyValue => keyValue.key === 'cause')
+        if (causeKeyValueFromExtensions) {
+          if (truncateCause) {
+            causeKeyValueFromExtensions.value = `${this.stack}\n${causeKeyValueFromExtensions.value}`.substring(ErrorEnums.MojaloopModelTypes.ExtensionValue.constraints.min - 1, ErrorEnums.MojaloopModelTypes.ExtensionValue.constraints.max) // truncate string to match Mojaloop API v1.0 Spec
+          } else {
+            causeKeyValueFromExtensions.value = `${this.stack}\n${causeKeyValueFromExtensions.value}`
+          }
+        } else {
+          let causeKeyValue
+          if (truncateCause) {
```

Comment:
This should be a direct Boolean comparison.  This could pass if any value other than `false` is used.

i.e.

```
if (truncateCause === true)
```